### PR TITLE
add more death shaming toggles

### PR DESCRIPTION
**Pull request by mr-d-rs**
add the following features
-ToB toggle (previously was always on)
-Toggle death shaming outside of raids
-Toggle death shaming for friends
-Toggle death shaming for strangers

### CODE REVIEW

File: src/main/java/ejedev/raidshamer/RaidShamerConfig.java [hunk 1]
```diff
@@ -10,43 +10,87 @@ public interface RaidShamerConfig extends Config
 {
 
     @ConfigItem(
-        keyName = "captureOwnDeaths",
-        name = "Own Deaths Captured",
-        description = "Allows you to toggle on/off own death shaming",
-        position = 1
+            keyName = "captureOwnDeaths",
```
Comment (jack0lantern) [inline src/main/java/ejedev/raidshamer/RaidShamerConfig.java:13]: Keep tabs to 4 spaces, same for all config items

Comment (mr-d-rs) [reply src/main/java/ejedev/raidshamer/RaidShamerConfig.java]: fixed in 5dfe6b6

**Comment by mr-d-rs** [comment]
@jack0lantern is it to your liking / ready for merge?

**Comment by jack0lantern** [comment]
Apologies for the delay, the code looks good. however, from a feature perspective, I'm not sure about "active in other areas". Based on the logic it seems it would strictly trigger when a player is outside raids. I could understand if someone wanted *all* areas to trigger an event, but could you explain the use case for excluding raids from that option?

anyway, I'm open to the idea, but consider rephrasing it - "Active outside of raids"

**Comment by jack0lantern** [comment]
Last thing, not sure about the use case of Stranger deaths. It seems odd to only want to capture and post stranger deaths. Consider getting rid of it, and default the friend deaths unchecked. currently, all deaths are sent by default and I don't want to change that for existing users when the update goes live.

I know this is a lotta input, thanks for the PR! Feel free to further discuss the feature additions before making changes, and if I can help with any changes

**Comment by mr-d-rs** [comment]
> Apologies for the delay, the code looks good. however, from a feature perspective, I'm not sure about "active in other areas". Based on the logic it seems it would strictly trigger when a player is outside raids. I could understand if someone wanted _all_ areas to trigger an event, but could you explain the use case for excluding raids from that option?
> 
> anyway, I'm open to the idea, but consider rephrasing it - "Active outside of raids"

The idea was to have all of the "location" toggles be disjoint. Raids are already covered by the other toggles. I will rename/rephrase it

**Comment by mr-d-rs** [comment]
> 



> Last thing, not sure about the use case of Stranger deaths. It seems odd to only want to capture and post stranger deaths. Consider getting rid of it, and default the friend deaths (rename "Friends' death only") unchecked. currently, all deaths are sent by default and I don't want to change that for existing users when the update goes live.
> 
> I know this is a lotta input, thanks for the PR! Feel free to further discuss the feature additions before making changes, and if I can help with any changes

this makes a lot of sense, will do. thanks for the feedback and no worries re: the delay

**Comment by mr-d-rs** [comment]
@jack0lantern 
the recent commits address each of your comments. For de7d223 though, there aren't any function or variable names that needed to be changed. Did you mean renaming the class and interface names / file names?

**Comment by jack0lantern** [comment]
> @jack0lantern the recent commits address each of your comments. For [de7d223](https://github.com/jack0lantern/raidshamer/commit/de7d2237479d027313a30710a5e26c7744ade217) though, there aren't any function or variable names that needed to be changed. Did you mean renaming the class and interface names / file names?

Yes, any reference to raidshamer should now be deathshamer

**Comment by mr-d-rs** [comment]
082c295 - raid shamer renamed to death shamer in all non-.gitignored places with the following exceptions
-support link in runelite-plugin.properties still points to https://github.com/jack0lantern/raidshamer
-the top level directory where the git repository lives is still named raidshamer

Only repository admins can rename the repository @jack0lantern

**Comment by jack0lantern** [comment]
I'll rename after merge. Running into issue where both "death shamer" and "raid shamer" show up in the plugin bar - not sure how to properly change the plugin name. The config, interestingly, is reset, but the old config seems to be in effect

**Comment by jack0lantern** [comment]
Sorry this fell by the wayside, so I asked the RL community about the above issue and here's the fix.

1. Revert the rename of the ConfigGroup
2. set the plugin descriptor configName ="RaidShamerPlugin"

This should resolve the issue

**Comment by jack0lantern** [comment]
@mr-d-rs if you don't make the changes in a week or so, or aren't able to, I'll go ahead and make these changes on my own branch and credit you

**Comment by jhayes45** [comment]
@jack0lantern sorry this fell by the wayside for me also. In 8059f5a the two fixes were applied. However this does not resolve the duplicate plugin issue in the plugin bar. It does make it so that both plugins share a config (if you toggle one of the options in one plugin it affects the options in both plugins) but both plugins still appear.

I looked around the github and discord and found that it is not recommended to rename the plugin files [1] [2]. So in ff8ee8b I renamed all of the source files back to raidshamer, keeping references to "Death Shamer" in the README and in the plugin descriptor. The result is that there is no duplicate plugin in the plugin bar anymore. Instead, there is just "Death Shamer" with all of the new config options.

[1] explicit guidance at https://github.com/runelite/plugin-hub/pull/4565#issuecomment-1615999652
[2] example of someone renaming their plugin https://github.com/runelite/plugin-hub/pull/3041/files - the source commit https://github.com/dey0/pluginhub-plugins/commit/bb7e3ae73c992774efa6ede9c9b58e0282fd16d9 only changes the name in the plugin descriptor

**Comment by mr-d-rs** [comment]
@jack0lantern sorry this fell by the wayside for me also. In https://github.com/jack0lantern/raidshamer/commit/8059f5a71bf95f3af86d83d4b59a4c37c02e73a6 the two fixes were applied. However this does not resolve the duplicate plugin issue in the plugin bar. It does make it so that both plugins share a config (if you toggle one of the options in one plugin it affects the options in both plugins) but both plugins still appear.

I looked around the github and discord and found that it is not recommended to rename the plugin files [1] [2]. So in https://github.com/jack0lantern/raidshamer/commit/ff8ee8b32e427cd0c7579126cd1947e339d0355e I renamed all of the source files back to raidshamer, keeping references to "Death Shamer" in the README and in the plugin descriptor. The result is that there is no duplicate plugin in the plugin bar anymore. Instead, there is just "Death Shamer" with all of the new config options.

[1] explicit guidance at https://github.com/runelite/plugin-hub/pull/4565#issuecomment-1615999652
[2] example of someone renaming their plugin https://github.com/runelite/plugin-hub/pull/3041/files - the source commit https://github.com/dey0/pluginhub-plugins/commit/bb7e3ae73c992774efa6ede9c9b58e0282fd16d9 only changes the name in the plugin descriptor

**Pull request merged**